raiden/tests/utils/transport.py

```diff
@@ -145,67 +139,76 @@ def __getattr__(self, item):
+                self.credentials = json.loads(credentials_file.read_text())
+            except (JSONDecodeError, UnicodeDecodeError, OSError) as ex:
+                raise AssertionError(
+                    f"Could not read credentials file '{credentials_file}': {ex}"
+                ) from ex
+        elif "admin_credentials" in config:
+            self.credentials = config["admin_credentials"]
+        else:
+            raise AssertionError(
+                "Either 'credentials_file' or 'admin_credentials' must be specified in "
+                "auth provider config."
+            )
 
         msg = "Keys 'username' and 'password' expected in credentials."
         assert "username" in self.credentials, msg
         assert "password" in self.credentials, msg
 
-    @defer.inlineCallbacks
-    def check_password(self, user_id: str, password: str):
+    async def check_password(self, user_id: str, password: str) -> bool:
         if not password:
             self.log.error("No password provided, user=%r", user_id)
-            defer.returnValue(False)
+            return False
 
         username = user_id.partition(":")[0].strip("@")
         if username == self.credentials["username"] and password == self.credentials["password"]:
             self.log.info("Logging in well known admin user")
-            user_exists = yield self.account_handler.check_user_exists(user_id)
+            user_exists = await self.account_handler.check_user_exists(user_id)
             if not user_exists:
                 self.log.info("First well known admin user login, registering: user=%r", user_id)
-                user_id = yield self.account_handler._hs.get_registration_handler().register_user(
+                await self.account_handler._hs.get_registration_handler().register_user(
                     localpart=username, admin=True
                 )
-                _, access_token = yield self.account_handler.register_device(user_id)
-                yield user_id, access_token
-            defer.returnValue(True)
-
-        self.log.error("Unknown user '%s', ignoring.", user_id)
-        defer.returnValue(False)
+            return True
+        return False
 
     @staticmethod
-    def parse_config(config):
+    def parse_config(config: Any) -> Any:
         return config
 
 
-# Used from within synapse during tests
 class EthAuthProvider:
     __version__ = "0.1"
     _user_re = re.compile(r"^@(0x[0-9a-f]{40}):(.+)$")
     _password_re = re.compile(r"^0x[0-9a-f]{130}$")
 
-    def __init__(self, config, account_handler):
+    def __init__(self, config, account_handler) -> None:  # type: ignore
         self.account_handler = account_handler
         self.config = config
         self.hs_hostname = self.account_handler._hs.hostname
         self.log = logging.getLogger(__name__)
 
-    @defer.inlineCallbacks
-    def check_password(self, user_id, password):
+    async def check_password(self, user_id: str, password: str) -> bool:
```

Comment:
Did synapse change away from twisted, or does twisted work with the async syntax, now?

Reply:
They are slowly converting to asyncio. 
This API has already been changed.

I'm not sure what they're using as a compatibility layer, but one of the twisted maintainers works for them so I assume they have that covered.